actionpack/test/controller/request_forgery_protection_test.rb

```diff
@@ -710,6 +714,46 @@ def test_rejects_token_for_incorrect_method
     end
   end
 
+  def test_rejects_token_for_incorrect_method_button_to
+    get :button_to, params: { form_method: 'delete' }
+
+    form_token = nil
+    assert_select 'input[name=custom_authenticity_token]' do |elts|
+      form_token = elts.first['value']
+      assert_not_nil form_token
+    end
+
+    actual = @controller.send(:unmask_token, Base64.strict_decode64(form_token))
+    expected = @controller.send(:per_form_csrf_token, session, '/per_form_tokens/post_one', 'delete')
+    assert_equal expected, actual
```

Comment:
Hard to track the purpose of these 8 lines comparing the generated token to the one on the server. Could do well with a custom assertion. E.g. `assert_matches_session_token form_token`. With that we'd need a helper to extract the `form_token` ala:

```ruby
def form_csrf_token
  assert_select 'input[name="custom_authenticity_token]" do |input|
    return input.first['value']
  end
end
```